bokehjs/src/lib/models/tools/inspectors/hover_tool.ts

```diff
@@ -1,81 +1,102 @@
 import {Template, TemplateView} from "../../dom"
+import {GlyphView} from "../../glyphs/glyph"
+import {HAreaView} from "../../glyphs/harea"
+import {LineView} from "../../glyphs/line"
+import {MultiLineView} from "../../glyphs/multi_line"
+import {PatchView} from "../../glyphs/patch"
+import {VAreaView} from "../../glyphs/varea"
+import {DataRenderer} from "../../renderers/data_renderer"
+import {GlyphRenderer} from "../../renderers/glyph_renderer"
+import {GraphRenderer} from "../../renderers/graph_renderer"
+import {Renderer} from "../../renderers/renderer"
+import {ImageIndex, Selection} from "../../selections/selection"
+import {ColumnarDataSource} from "../../sources/columnar_data_source"
+import {compute_renderers} from "../../util"
+import {CustomJSHover} from "./customjs_hover"
+import {InspectTool, InspectToolView} from "./inspect_tool"
 
-export type TooltipVars = {index?: number} & Vars
-
-export function _nearest_line_hit(i: number, geometry: Geometry,
-    sx: number, sy: number, dx: Arrayable<number>, dy: Arrayable<number>): [[number, number], number] {
-  const d1 = {x: dx[i], y: dy[i]}
-  const d2 = {x: dx[i+1], y: dy[i+1]}
-
-  let dist1: number
-  let dist2: number
-  if (geometry.type == "span") {
-    if (geometry.direction == "h") {
-      dist1 = Math.abs(d1.x - sx)
-      dist2 = Math.abs(d2.x - sx)
-    } else {
-      dist1 = Math.abs(d1.y - sy)
-      dist2 = Math.abs(d2.y - sy)
+export type TooltipVars = {
+    index: number | null
+    glyph: GlyphView
+    x: number
+    y: number
+    sx: number
+    sy: number
+    tt_x: number
+    tt_y: number
+    tt_sx: number
+    tt_sy: number
+    name: string | null
+    indices?: any | undefined
+    segment_index?: any | undefined
+    image_index?: ImageIndex | undefined
+}
+
+export function _nearest_line_hit(
+  i: number,
+  geometry: PointGeometry | SpanGeometry,
+  dx: Arrayable<number>, dy: Arrayable<number>
+): [[number, number], number] {
+
+  const p1 = {x: dx[i], y: dy[i]}
+  const p2 = {x: dx[i+1], y: dy[i+1]}
+  const {sx, sy} = geometry
+
+  const [d1, d2] = (function() {
+    if (geometry.type == "span") {
+      if (geometry.direction == "h")
+        return [Math.abs(p1.x - sx), Math.abs(p2.x - sx)]
+      else
+        return [Math.abs(p1.y - sy), Math.abs(p2.y - sy)]
     }
-  } else {
+
+    // point geometry case
     const s = {x: sx, y: sy}
-    dist1 = hittest.dist_2_pts(d1, s)
-    dist2 = hittest.dist_2_pts(d2, s)
-  }
+    const d1 = hittest.dist_2_pts(p1, s)
+    const d2 = hittest.dist_2_pts(p2, s)
+    return [d1, d2]
+  })()
 
-  if (dist1 < dist2)
-    return [[d1.x, d1.y], i]
-  else
-    return [[d2.x, d2.y], i+1]
+  return d1 < d2 ? [[p1.x, p1.y], i] : [[p2.x, p2.y], i+1]
 }
 
-export function _line_hit(xs: Arrayable<number>, ys: Arrayable<number>, ind: number): [[number, number], number] {
-  return [[xs[ind], ys[ind]], ind]
+export function _line_hit(
+  xs: Arrayable<number>,
+  ys: Arrayable<number>,
+  i: number
```

Comment:
If you insist on rewriting those signatures, at least use a trailing comma.

Reply:
[60556ad](https://github.com/bokeh/bokeh/pull/12147/commits/60556ad9f4b1a9a25bf12298df52dccc99ded769)